tools/auto-fuzz/manager.py

```diff
@@ -196,13 +198,71 @@ def run_static_analysis_python(git_repo, basedir):
     return ret
 
 
+def _maven_build_project(basedir, projectdir):
+    """Helper method to build project using maven"""
+    # Prepare maven
+    with zipfile.ZipFile(os.path.join(basedir, "maven.zip"), "r") as mf:
+        mf.extractall(basedir)
+
+    # Set environment variable
+    env_var = os.environ.copy()
+    env_var['PATH'] = os.path.join(basedir, "apache-maven-3.6.3",
+                                   "bin") + ":" + env_var['PATH']
+
+    # Build project with maven
+    cmd = [
+        "mvn clean package", "-DskipTests", "-Djavac.src.version=15",
+        "-Djavac.target.version=15", "-Dmaven.javadoc.skip=true"
+    ]
+    try:
+        subprocess.check_call(" ".join(cmd),
+                              shell=True,
+                              timeout=1800,
+                              stdout=subprocess.DEVNULL,
+                              stderr=subprocess.DEVNULL,
+                              env=env_var,
+                              cwd=projectdir)
+    except subprocess.TimeoutExpired:
+        pass
+
+
+def _gradle_build_project(basedir, projectdir):
+    """Helper method to build project using maven"""
+    # Prepare gradle
+    with zipfile.ZipFile(os.path.join(basedir, "gradle.zip"), "r") as gf:
+        gf.extractall(basedir)
+
+    # Set environment variable
+    env_var = os.environ.copy()
+    env_var['GRADLE_HOME'] = os.path.join(basedir, "gradle-7.4.2")
+    env_var['PATH'] = os.path.join(basedir, "gradle-7.4.2",
+                                   "bin") + ":" + env_var['PATH']
+
+    # Build project with maven
+    cmd = [
+        "./gradlew clean build -x test",
+        "jar cvf proj.jar -C build/classes/java/main/ ."
+    ]
+    try:
+        subprocess.check_call(" && ".join(cmd),
+                              shell=True,
+                              timeout=1800,
+                              stdout=subprocess.DEVNULL,
+                              stderr=subprocess.DEVNULL,
+                              env=env_var,
+                              cwd=projectdir)
+    except subprocess.TimeoutExpired:
```

Comment:
I've had troubles where this breaks build because this command fails. We would like to avoid some commands failing causing a large-scale experiment breaking.